applications/vanilla/models/class.discussionmodel.php

```diff
@@ -2634,7 +2633,21 @@ public function updateDiscussionCount($categoryID, $discussion = false) {
             $this->Database->query($sql, $params, 'DiscussionModel_UpdateDiscussionCount');
 
         } elseif (is_numeric($categoryID)) {
-            $this->SQL
+            /** @var Vanilla\Scheduler\SchedulerInterface $scheduler */
+            $scheduler = Gdn::getContainer()->get(Vanilla\Scheduler\SchedulerInterface::class);
+            $scheduler->addJob(Vanilla\Library\Jobs\UpdateDiscussionCount::class, ['categoryID' => $categoryID, 'discussion' => $discussion]);
+        }
+    }
+
+    /**
+     * Scheduled category discussion count update.
+     *
+     * @param int $categoryID Unique ID of category we are updating.
+     * @param array|false $discussion The discussion to update the count for or **false** for all of them.
+     */
+    public function scheduledUpdateCount($categoryID, $discussion = false) {
```

Comment:
The method is `scheduledUpdateCount`, but I wouldn't think this had anything to do with a scheduled job based on its content. It just looks like a method I'd use to update a category's discussion and comment fields.

Reply:
I normally wouldn't nitpick, but since this is new: it feels like a method for updating category fields should be in the category model.

Reply:
When refactoring old code like this into new methods, it's a good chance to improve on them.

1. We probably wouldn't allow `false` as a value for an array parameter, anymore. If it's truly optional, we can make it nullable.
1. We can use types for parameters. `$discussion` looks like it already assumes it'll be an array, if it's anything. `$categoryID` will only work as an integer.

It's probably a good idea to alter usage at the call site (or, better still, when you're scheduling the job) to account for the signature changes in the method. Since it's currently only really used in one place, it should be easy.